extension/src/PermissionSet/XmlPermissionSet.ts

```diff
@@ -27,35 +29,51 @@ export class XmlPermissionSets {
         suggestedNewName: prefix + roleId,
       };
-      const permissionsNodeList = permissionSetNode.getElementsByTagName(
+      let permissionsNodeList = permissionSetNode.getElementsByTagName(
         "Permission"
       );
+      if (permissionsNodeList.length === 0) {
+        permissionsNodeList = permissionSetNode.getElementsByTagName(
+          "TenantPermission"
+        );
+      }
       for (let i = 0; i < permissionsNodeList.length; i++) {
         const permissionsNode = permissionsNodeList[i];
-        const objectType = permissionsNode.getElementsByTagName("ObjectType")[0]
-          .childNodes[0].nodeValue;
-
-        const objectId = permissionsNode.getElementsByTagName("ObjectID")[0]
-          .childNodes[0].nodeValue;
-        const readPermission =
-          permissionsNode.getElementsByTagName("ReadPermission")[0]
-            .childNodes[0].nodeValue ?? "";
-        const insertPermission =
-          permissionsNode.getElementsByTagName("InsertPermission")[0]
-            .childNodes[0].nodeValue ?? "";
-        const modifyPermission =
-          permissionsNode.getElementsByTagName("ModifyPermission")[0]
-            .childNodes[0].nodeValue ?? "";
-        const deletePermission =
-          permissionsNode.getElementsByTagName("DeletePermission")[0]
-            .childNodes[0].nodeValue ?? "";
-        const executePermission =
-          permissionsNode.getElementsByTagName("ExecutePermission")[0]
-            .childNodes[0].nodeValue ?? "";
+        const objectType = XmlPermissionSets.getPermissionValue(
+          permissionsNode,
+          "ObjectType"
+        );
+        const objectId = XmlPermissionSets.getPermissionValue(
+          permissionsNode,
+          "ObjectID"
+        );
+        const readPermission = XmlPermissionSets.getPermissionValue(
+          permissionsNode,
+          "ReadPermission"
+        );
+        const insertPermission = XmlPermissionSets.getPermissionValue(
+          permissionsNode,
+          "InsertPermission"
+        );
+        const modifyPermission = XmlPermissionSets.getPermissionValue(
+          permissionsNode,
+          "ModifyPermission"
+        );
+        const deletePermission = XmlPermissionSets.getPermissionValue(
+          permissionsNode,
+          "DeletePermission"
+        );
+        const executePermission = XmlPermissionSets.getPermissionValue(
+          permissionsNode,
+          "ExecutePermission"
+        );
+        const securityFilter = permissionsNode.getElementsByTagName(
+          "SecurityFilter"
+        );
         if (
-          permissionsNode
-            .getElementsByTagName("SecurityFilter")[0]
-            .hasChildNodes()
+          securityFilter &&
+          securityFilter[0] &&
+          securityFilter[0].hasChildNodes()
         ) {
           throw new Error(
             `PermissionSet ${permissionSet.roleID} har defined some SecurityFilter, which is unsupported by this function`
```

Comment:
Swenglish? 🙃

Reply:
Ooops, I thought I changed that already... 😖